[WPE] WPE Platform: change the API to get the preferred DMA-BUF formats

carlosgcampos opened the pull request:
#### f78822d601b744ee397c5b655c3d9191498e5821
<pre>
[WPE] WPE Platform: change the API to get the preferred DMA-BUF formats
<a href="https://bugs.webkit.org/show_bug.cgi?id=272667">https://bugs.webkit.org/show_bug.cgi?id=272667</a>

Reviewed by Alejandro G. Castro.

The initial API we added was very simple, just providing a single list
of formats, usages and modifiers, but it turned out it was not so simple
to iterate the formats, since we always want to iterate by tranche.
Also, we want to handle the main and target devices in the future, so we
need a new API that simplifies the iteration by tranche and includes the
devices.

* Source/WebKit/Shared/glib/DMABufRendererBufferFormat.h:
(WebKit::DMABufRendererBufferFormat::operator== const): Deleted.
* Source/WebKit/Shared/glib/DMABufRendererBufferFormat.serialization.in:
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
(WebKit::WebPageProxy::preferredBufferFormats const):
* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABufFormat.cpp: Removed.
* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABufFormat.h: Removed.
* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABufFormats.cpp: Added.
(DMABufFormat::DMABufFormat):
(DMABufFormatsGroup::DMABufFormatsGroup):
(wpe_buffer_dma_buf_formats_class_init):
(wpe_buffer_dma_buf_formats_get_device):
(wpe_buffer_dma_buf_formats_get_n_groups):
(wpe_buffer_dma_buf_formats_get_group_usage):
(wpe_buffer_dma_buf_formats_get_group_device):
(wpe_buffer_dma_buf_formats_get_group_n_formats):
(wpe_buffer_dma_buf_formats_get_format_fourcc):
(wpe_buffer_dma_buf_formats_get_format_modifiers):
(_WPEBufferDMABufFormatsBuilder::_WPEBufferDMABufFormatsBuilder):
(wpe_buffer_dma_buf_formats_builder_new):
(wpe_buffer_dma_buf_formats_builder_ref):
(wpe_buffer_dma_buf_formats_builder_unref):
(wpe_buffer_dma_buf_formats_builder_append_group):
(wpe_buffer_dma_buf_formats_builder_append_format):
(wpe_buffer_dma_buf_formats_builder_end):
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
(wpeDisplayDispose):
(wpeDisplayPreferredDMABufFormats):
(wpe_display_get_preferred_dma_buf_formats):
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.h:
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpe_view_class_init):
(wpe_view_get_preferred_dma_buf_formats):
(wpeViewDispose): Deleted.
* Source/WebKit/WPEPlatform/wpe/WPEView.h:
* Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
(wpeDisplayDRMGetPreferredDMABufFormats):
* Source/WebKit/WPEPlatform/wpe/headless/WPEDisplayHeadless.cpp:
(wpeDisplayHeadlessGetPreferredDMABufFormats):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
(wpeDisplayWaylandGetPreferredDMABufFormats):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp:
(DMABufFeedback::DMABufFeedback):
(DMABufFeedback::drmDeviceForUsage):
(DMABufFeedback::drmDevice const):
(DMABufFeedback::Tranche::Tranche):
(DMABufFeedback::Tranche::drmDevice const):
(wpeViewWaylandGetPreferredDMABufFormats):
* Source/WebKit/WPEPlatform/wpe/wpe-platform.h:
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp:
(WebKit::AcceleratedSurfaceDMABuf::RenderTargetEGLImage::create):
(WebKit::AcceleratedSurfaceDMABuf::SwapChain::setupBufferFormat):
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.h:

Canonical link: <a href="https://commits.webkit.org/277538@main">https://commits.webkit.org/277538@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/08fb2b3bb53d1f109c48406f9b8df276b805a185

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47943 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27146 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50822 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50624 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43996 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33018 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24633 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/39004 "Failure limit exceed. At least found 1 new test failure: fast/css-custom-paint/out-of-memory-while-adding-worklet-module.html (failure)") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48525 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24817 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/41471 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20304 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/22293 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5990 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/44289 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/43089 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52518 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22986 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/19335 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/46312 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24255 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45358 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25047 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6788 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23977 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alexgcastro reviewed (approved):
LGTM! Great addition.

carlosgcampos committed:
[WPE] WPE Platform: change the API to get the preferred DMA-BUF formats
https://bugs.webkit.org/show_bug.cgi?id=272667

Reviewed by Alejandro G. Castro.

The initial API we added was very simple, just providing a single list
of formats, usages and modifiers, but it turned out it was not so simple
to iterate the formats, since we always want to iterate by tranche.
Also, we want to handle the main and target devices in the future, so we
need a new API that simplifies the iteration by tranche and includes the
devices.

* Source/WebKit/Shared/glib/DMABufRendererBufferFormat.h:
(WebKit::DMABufRendererBufferFormat::operator== const): Deleted.
* Source/WebKit/Shared/glib/DMABufRendererBufferFormat.serialization.in:
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
(WebKit::WebPageProxy::preferredBufferFormats const):
* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABufFormat.cpp: Removed.
* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABufFormat.h: Removed.
* Source/WebKit/WPEPlatform/wpe/WPEBufferDMABufFormats.cpp: Added.
(DMABufFormat::DMABufFormat):
(DMABufFormatsGroup::DMABufFormatsGroup):
(wpe_buffer_dma_buf_formats_class_init):
(wpe_buffer_dma_buf_formats_get_device):
(wpe_buffer_dma_buf_formats_get_n_groups):
(wpe_buffer_dma_buf_formats_get_group_usage):
(wpe_buffer_dma_buf_formats_get_group_device):
(wpe_buffer_dma_buf_formats_get_group_n_formats):
(wpe_buffer_dma_buf_formats_get_format_fourcc):
(wpe_buffer_dma_buf_formats_get_format_modifiers):
(_WPEBufferDMABufFormatsBuilder::_WPEBufferDMABufFormatsBuilder):
(wpe_buffer_dma_buf_formats_builder_new):
(wpe_buffer_dma_buf_formats_builder_ref):
(wpe_buffer_dma_buf_formats_builder_unref):
(wpe_buffer_dma_buf_formats_builder_append_group):
(wpe_buffer_dma_buf_formats_builder_append_format):
(wpe_buffer_dma_buf_formats_builder_end):
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.cpp:
(wpeDisplayDispose):
(wpeDisplayPreferredDMABufFormats):
(wpe_display_get_preferred_dma_buf_formats):
* Source/WebKit/WPEPlatform/wpe/WPEDisplay.h:
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpe_view_class_init):
(wpe_view_get_preferred_dma_buf_formats):
(wpeViewDispose): Deleted.
* Source/WebKit/WPEPlatform/wpe/WPEView.h:
* Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
(wpeDisplayDRMGetPreferredDMABufFormats):
* Source/WebKit/WPEPlatform/wpe/headless/WPEDisplayHeadless.cpp:
(wpeDisplayHeadlessGetPreferredDMABufFormats):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
(wpeDisplayWaylandGetPreferredDMABufFormats):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp:
(DMABufFeedback::DMABufFeedback):
(DMABufFeedback::drmDeviceForUsage):
(DMABufFeedback::drmDevice const):
(DMABufFeedback::Tranche::Tranche):
(DMABufFeedback::Tranche::drmDevice const):
(wpeViewWaylandGetPreferredDMABufFormats):
* Source/WebKit/WPEPlatform/wpe/wpe-platform.h:
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp:
(WebKit::AcceleratedSurfaceDMABuf::RenderTargetEGLImage::create):
(WebKit::AcceleratedSurfaceDMABuf::SwapChain::setupBufferFormat):
* Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.h:

Canonical link: https://commits.webkit.org/277538@main
